--- CMakeLists.txt.orig	2017-11-01 09:40:08.810173755 +0100
+++ CMakeLists.txt	2017-11-01 09:40:28.978720023 +0100
@@ -39,6 +39,14 @@
 	ENDIF()
 ENDMACRO(require)
 
+get_property(LIB64 GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS)
+
+if ("${LIB64}" STREQUAL "TRUE")
+    set(LIBSUFFIX 64)
+else()
+    set(LIBSUFFIX "")
+endif()
+
 # ** When a C++11 feature is introduced into the code, uncomment the equiviliant line so ensure compatability. **
 # require(HAS_CXX11_LONG_LONG "C++11 lambdas\t\t\t")
 # require(HAS_CXX11_AUTO "C++11 auto support:\t\t\t")
--- ancillary/configure/api/downloader-api/CMakeLists.txt.orig	2017-11-01 09:43:08.736125451 +0100
+++ ancillary/configure/api/downloader-api/CMakeLists.txt	2017-11-01 09:43:27.450704318 +0100
@@ -65,7 +65,7 @@
 
 INSTALL(TARGETS xfstk-dldr-gui-api
     LIBRARY
-        DESTINATION /usr/lib
+        DESTINATION lib${LIBSUFFIX}
         NAMELINK_SKIP
         COMPONENT lib)
         
--- ancillary/configure/plugins/downloader/baytrail/CMakeLists.txt.orig	2017-11-01 09:44:58.059665225 +0100
+++ ancillary/configure/plugins/downloader/baytrail/CMakeLists.txt	2017-11-01 09:45:22.511114927 +0100
@@ -53,5 +53,5 @@
         AUTOMOC true)
 
 INSTALL(TARGETS XfstkDldrPluginBaytrail
-    LIBRARY DESTINATION /usr/lib/xfstk/xfstkdldrplugins
+    LIBRARY DESTINATION /usr/lib${LIBSUFFIX}/xfstk/xfstkdldrplugins
         COMPONENT gui)
--- ancillary/configure/plugins/downloader/cloverview/CMakeLists.txt.orig	2017-11-01 09:46:08.294084512 +0100
+++ ancillary/configure/plugins/downloader/cloverview/CMakeLists.txt	2017-11-01 09:46:18.882846196 +0100
@@ -55,5 +55,5 @@
         AUTOMOC true)
 
 INSTALL(TARGETS XfstkDldrPluginCloverview
-    LIBRARY DESTINATION /usr/lib/xfstk/xfstkdldrplugins
+    LIBRARY DESTINATION /usr/lib${LIBSUFFIX}/xfstk/xfstkdldrplugins
         COMPONENT gui)
--- ancillary/configure/plugins/downloader/cloverviewplus/CMakeLists.txt.orig	2017-11-01 09:46:59.140940088 +0100
+++ ancillary/configure/plugins/downloader/cloverviewplus/CMakeLists.txt	2017-11-01 09:47:15.429573464 +0100
@@ -55,6 +55,6 @@
         AUTOMOC true)
 
 INSTALL(TARGETS XfstkDldrPluginCloverviewPlus
-        LIBRARY DESTINATION /usr/lib/xfstk/xfstkdldrplugins
+        LIBRARY DESTINATION /usr/lib${LIBSUFFIX}/xfstk/xfstkdldrplugins
         COMPONENT gui)
 
--- ancillary/configure/plugins/downloader/medfield/CMakeLists.txt.orig	2017-11-01 09:48:02.907504807 +0100
+++ ancillary/configure/plugins/downloader/medfield/CMakeLists.txt	2017-11-01 09:48:19.778125065 +0100
@@ -60,5 +60,5 @@
         AUTOMOC true)
                             
 INSTALL(TARGETS XfstkDldrPluginMedfield
-    LIBRARY DESTINATION /usr/lib/xfstk/xfstkdldrplugins
+    LIBRARY DESTINATION /usr/lib${LIBSUFFIX}/xfstk/xfstkdldrplugins
         COMPONENT gui)
--- ancillary/configure/plugins/downloader/merrifield/CMakeLists.txt.orig	2017-11-01 09:49:22.092722384 +0100
+++ ancillary/configure/plugins/downloader/merrifield/CMakeLists.txt	2017-11-01 09:55:02.150067051 +0100
@@ -53,5 +53,5 @@
         AUTOMOC true)
 
 INSTALL(TARGETS XfstkDldrPluginMerrifield
-    LIBRARY DESTINATION /usr/lib/xfstk/xfstkdldrplugins
+    LIBRARY DESTINATION /usr/lib${LIBSUFFIX}/xfstk/xfstkdldrplugins
         COMPONENT gui)
